ipol_demo/modules/blobs/blobs.py

```diff
@@ -28,1609 +27,1217 @@
-        if vr_file:
-            os.remove(vr_file[0])
-            self.remove_dirs(vr_folder)
-
-    def remove_dirs(self, blob_folder):
-        """
-        Remove all the empty directories
-        """
-        try:
-            if not os.listdir(blob_folder):
-                os.rmdir(blob_folder)
-                self.remove_dirs(os.path.dirname(blob_folder))
-        except Exception as ex:
-            raise IPOLRemoveDirError(ex)
-
-    @staticmethod
-    def generate_set_name(blob_id):
-        """
-        Generate a unique set name for the given blob id
-        """
-        return "__" + str(blob_id)
-
-    @cherrypy.expose
-    @authenticate
-    def edit_blob_from_demo(
-        self,
-        demo_id=None,
-        blob_set=None,
-        new_blob_set=None,
-        pos_set=None,
-        new_pos_set=None,
-        title=None,
-        credit=None,
-        vr=None,
-    ):
-        """
-        Edit blob information in a demo
-        """
-        dest = {"dest": "demo", "demo_id": demo_id}
-        if self.edit_blob(
-            blob_set, new_blob_set, pos_set, new_pos_set, title, credit, vr, dest
-        ):
-            return json.dumps({"status": "OK"}).encode()
-        return json.dumps({"status": "KO"}).encode()
-
-    @cherrypy.expose
-    @authenticate
-    def edit_blob_from_template(
-        self,
-        template_id=None,
-        blob_set=None,
-        new_blob_set=None,
-        pos_set=None,
-        new_pos_set=None,
-        title=None,
-        credit=None,
-        vr=None,
-    ):
-        """
-        Edit blob information in a template
-        """
-        dest = {"dest": "template", "id": template_id}
-        if self.edit_blob(
-            blob_set, new_blob_set, pos_set, new_pos_set, title, credit, vr, dest
-        ):
-            return json.dumps({"status": "OK"}).encode()
-        return json.dumps({"status": "KO"}).encode()
-
-    def edit_blob(
-        self, blob_set, new_blob_set, pos_set, new_pos_set, title, credit, blob_vr, dest
-    ):
-        """
-        Edit blob information
-        """
+def get_vr_extension(vr_dir: str, blob_hash: str) -> str:
+    """
+    If the visual representation exists, the function returns its extension
+    if not, returns None
+    """
+    vr_extension = None
+    if os.path.isdir(vr_dir):
+        file_without_extension = os.path.join(vr_dir, blob_hash)
+        vr = glob.glob(file_without_extension + ".*")
+        if vr:
+            _, vr_extension = os.path.splitext(vr[0])
+
+    return vr_extension
+
+
+@app.get("/demo_templates/{demo_id}", status_code=200)
+def get_demo_templates(demo_id: int) -> list:
+    """
+    Get the list of templates used by the demo
+    """
+    with get_db_connection() as conn:
+        return database.get_demo_templates(conn, demo_id)
+
+
+def remove_blob(blob_set: str, pos_set: int, dest: str) -> bool:
+    """
+    Remove the blob
+    """
+    with Lock():
```

Comment:
This creates a new instance of lock at each call, which means it's not locking anything.